plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/condition/BrowserContainsText.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat, Inc and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.reddeer.eclipse.condition;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.reddeer.common.condition.AbstractWaitCondition;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.swt.condition.PageIsLoaded;
+import org.eclipse.reddeer.swt.impl.browser.InternalBrowser;
+
+/**
+ * Wait condition which returns true if a given browser has the specified URL and specific text.
+ */
+public class BrowserContainsText extends AbstractWaitCondition {
+
+	private InternalBrowser internalBrowser;
+	private String text;
+	
+	public BrowserContainsText(InternalBrowser browser, String text) {
+		this.internalBrowser = browser;
+		this.text = text;
+	}
+
+	public BrowserContainsText(String text) {
+		this(new InternalBrowser(), text);
+	}
+
+	@Override
+	public boolean test() {
+		internalBrowser = new InternalBrowser();
```

Comment:
this will always override browser specified by user. Also we should require Browser instead of InternalBrowser